google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiManager.java

```diff
@@ -87,6 +88,10 @@
   private static final String SERVICE_ACCOUNT_ROLE_REQUEST_PREFIX = "serviceAccount:";
   private static final String SERVICE_ACCOUNT_KEY_FILE_NAME_FORMAT = "%s-%s.json";
 
+  private static final int SERVICE_ACCOUNT_ID_MAX_LEN = 30;
+  static final int SERVICE_ACCOUNT_NAME_MAX_LEN = 100;
```

Comment:
Could you please expand a bit on the difference between ID and NAME length here and their use in validation?

Reply:
Sure: these are 2 separate components of a Service Account. It has a name (which doesn't need to be unique, can be up to 100 characters, and can have any characters). The id is restricted to 30 chars, must be unique (hence the timestamp), and is subject to the regex in this PR. Does that answer the question?

Reply:
Yes, thanks.
So if I understand correctly: we allow name to be at max 100 chars, then check for valid characters, then format it as `name-timestamp`, and then finally trim that to 30 chars?

Reply:
oh. i forgot to mention: the name is used as the seed for the id. So the name is maxed at 100. But then the ID is created based off of the name (capped at 30 characters including a timestamp that is added).

so yes, you are correct.

Reply:
So it looks like minus the timestamp, we only include ~20 chars from the original name into ID, correct? But we still use the name for the service account for presentation purposes, i.e. its visible in GCP Console?

Reply:
yes. I'm also following the strategy used by the web console (of including the filtered name as part of the id). We could potentially not include the name and bypass all of this regex validation mess.

Reply:
Ok, got it, thanks, I think it's still good to keep at least part of name in the ID - probably much easier to identify later.

Reply:
Yeah. I'll update the user-visible message to not use the regex and we'll leave in this naming strategy for now.